frontend/test/resolution/testTuples.cpp

```diff
@@ -846,6 +845,262 @@ static void testTupleGeneric() {
   argHelper("(numeric, numeric)", "(5, 'hi')", false);
 }
 
+static void test21() {
+  printf("%s\n", __FUNCTION__);
+  Context ctx;
+  auto context = &ctx;
+  ErrorGuard guard(context);
+
+  std::string program =
+    R""""(
+    operator=(ref lhs: real, ref rhs: real) {}
+    operator=(ref lhs: int, ref rhs: int) {}
+    proc foo() { return (1.0, 2); }
+    var x: real;
+    (x, _) = foo();
+    )"""";
+
+  auto xQt = resolveQualifiedTypeOfX(context, program);
+  assert(!guard.realizeErrors());
+  assert(xQt.kind() == QualifiedType::VAR);
+  assert(xQt.type()->isRealType());
+
+  // Get the type of the '(x, _)' tuple itself.
+  auto mod = parseModule(context, program);
+  auto& rr = resolveModule(context, mod->id());
+  auto astTup = mod->stmt(4)->toCall()->actual(0);
+  assert(astTup->isTuple());
+  auto& qtTup = rr.byAst(astTup).type();
+  assert(qtTup.kind() == QualifiedType::CONST_VAR);
+  assert(qtTup.type()->isTupleType());
```

Comment:
You should probably be checking the component kinds too, since your code adjusts them one-by-one.